otter/convergence/gathering.py

```diff
@@ -172,57 +171,46 @@ def get_scaling_group_servers(tenant_id, group_id, now,
     yield do_return(servers)
 
 
+@do
 def get_clb_contents():
-    """
-    Get Rackspace Cloud Load Balancer contents as list of `CLBNode`.
-    """
-
-    def lb_req(method, url, json_response=True):
-        """Make a request to the LB service with retries."""
-        return retry_effect(
-            service_request(
-                ServiceType.CLOUD_LOAD_BALANCERS,
-                method, url, json_response=json_response),
-            retry_times(5), exponential_backoff_interval(2))
-
-    def _lb_path(lb_id):
-        """Return the URL path to lb with given id's nodes."""
-        return append_segments('loadbalancers', str(lb_id), 'nodes')
-
-    def fetch_nodes(result):
-        _response, body = result
-        lbs = body['loadBalancers']
-        lb_ids = [lb['id'] for lb in lbs]
-        lb_reqs = [
-            lb_req('GET', _lb_path(lb_id)).on(
-                lambda (response, body): body['nodes'])
-            for lb_id in lb_ids]
-        return parallel(lb_reqs).on(lambda all_nodes: (lb_ids, all_nodes))
-
-    def fetch_drained_feeds((ids, all_lb_nodes)):
-        nodes = [
-            CLBNode.from_node_json(_id, node)
-            for _id, nodes in zip(ids, all_lb_nodes) for node in nodes]
-        draining = [n for n in nodes
-                    if n.description.condition == CLBNodeCondition.DRAINING]
-        return parallel(
-            [lb_req(
-                'GET',
-                append_segments(
-                    'loadbalancers',
-                    str(n.description.lb_id),
-                    'nodes',
-                    '{}.atom'.format(n.node_id)),
-                json_response=False).on(_discard_response)
-             for n in draining]).on(lambda feeds: (nodes, draining, feeds))
-
-    def fill_drained_at((nodes, draining, feeds)):
-        for node, feed in zip(draining, feeds):
-            node.drained_at = extract_CLB_drained_at(feed)
-        return nodes
-
-    return lb_req('GET', 'loadbalancers').on(
-        fetch_nodes).on(fetch_drained_feeds).on(fill_drained_at)
+    """Get Rackspace Cloud Load Balancer contents as list of `CLBNode`."""
+    # If we get a CLBNotFoundError while fetching feeds, we should throw away
+    # all nodes related to that load balancer, because we don't want to act on
+    # data that we know is invalid/outdated (for example, if we can't fetch a
+    # feed because CLB was deleted, we don't want to say that we have a node in
+    # DRAINING with draining time of 0; we should just say that the node is
+    # gone).
+
+    def gone(r):
+        return catch(CLBNotFoundError, lambda exc: r)
```

Comment:
Non-blocking - we can always file an issue or something - do we also want to catch `NoSuchCLBNodeError` here, in case the node itself (but not the CLB) is gone?  This would only probably affect the atom feed, and tbh I don't know if that's the exact error for getting the atom feed of a node.